azavea/raster-foundry
# Feature/add landsat8 footprints

This pull request includes all the functionality required to extract
footprints from landsat8 data. It accomplishes via the following
workflow:
 - downloads a landsat scene from s3
 - creates a tif mask of where 0 is nodata and 1 is data
 - extracts polygon from tif mask

Functions to get and post information to raster foundry via API
endpoints are roughly stubbed out, but will need to be adjusted once the
exact workflow of landsat 8 ingestion is decided.

Some additional work will be required once the exact workflows are designed, but should be minimal.

## Testing

1. Start VM
2. Run `./scripts/bootstrap` and `./scripts/setup` to install new dependencies
3. Run `./scripts/test` and ensure that test suite still passes (should be additional tests for `rf` python library)
4. Start a bash console on the `airflow-webserver`: `./scripts/console airflow-webserver bash`
5. Change directory: `/opt/raster-foundry/app-tasks/rf`
6. Start an `ipython` shell: `ipython`
7. Run the following commands to run through the process of extracting a polygon:
    ```python
import json
import os
from rf.uploads.landsat8.footprint import (
        download_tif,
        create_tif_mask,
        extract_polygon
)

temp_dir = './'
scene = 'LC81390452014295LGN00'
bands = [1]

tifs = download_tif(temp_dir, scene, bands)
first_tif = tifs.get(1)
print 'FIRST TIF LOCATED AT: {}'.format(first_tif)

mask_tif_path = create_tif_mask(temp_dir, first_tif)
print 'MASK TIF LOCATED AT: {}'.format(mask_tif_path)

geojson = extract_polygon(mask_tif_path)
with open(os.path.join(temp_dir, 'test.geojson'), 'wb') as fh:
        fh.write(json.dumps(geojson))
```
8. Verify tifs and polygon in qgis:
 
 a. First Tif (raw landsat data)
![raw-landsat](https://cloud.githubusercontent.com/assets/898060/18280315/a424b4dc-7426-11e6-955f-8d8f35ba14bd.png)
 b. Second tif (mask of polygon -- hard to see due to default qgis style)
![mask-landsat](https://cloud.githubusercontent.com/assets/898060/18280335/bb7ac518-7426-11e6-8dee-23ef3136a444.png)
 c. Polygon mask on top of second tif
![polygon](https://cloud.githubusercontent.com/assets/898060/18280353/c5390a38-7426-11e6-817c-4dba1d67c4d2.png)
